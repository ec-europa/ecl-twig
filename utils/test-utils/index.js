const Twig = require('twig');
const { format } = require('prettier');

const renderTwigFile = (file, options, cb) =>
  Twig.renderFile(file, options, (err, html) =>
    cb(err, format(html, { parser: 'html' }))
  );

const renderTwigFileAsNode = (file, options) =>
  new Promise((resolve, reject) =>
    Twig.renderFile(file, options, (err, html) => {
      if (err) return reject(err);

      const div = document.createElement('div');
      div.innerHTML = html.trim();

      return resolve(div.firstChild);
    })
  );

module.exports = {
  renderTwigFile,
  renderTwigFileAsNode,
};
