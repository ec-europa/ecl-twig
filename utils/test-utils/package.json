{
  "name": "@ecl-twig/test-utils",
  "version": "2.29.0",
  "description": "A local test helper",
  "main": "index.js",
  "private": true,
  "devDependencies": {
    "deepmerge": "4.2.2"
  }
}
