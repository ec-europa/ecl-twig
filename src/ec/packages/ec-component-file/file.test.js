import { merge, renderTwigFileAsNode } from '@ecl-twig/test-utils';

// Import data for tests
import { dataWithTranslation, dataWithoutTranslation } from './demo/data';

describe('EC - File', () => {
  const template = '@ecl-twig/ec-component-file/ecl-file.html.twig';
  const render = params => renderTwigFileAsNode(template, params);

  describe('With translation', () => {
    test('renders correctly', () => {
      expect.assertions(1);

      return expect(render(dataWithTranslation)).resolves.toMatchSnapshot();
    });

    test('renders correctly with extra class names', () => {
      expect.assertions(1);

      const withExtraClasses = merge(dataWithTranslation, {
        extra_classes: 'custom-class custom-class--test',
      });

      return expect(render(withExtraClasses)).resolves.toMatchSnapshot();
    });

    test('renders correctly with extra attributes', () => {
      expect.assertions(1);

      const withExtraAttributes = merge(dataWithTranslation, {
        extra_attributes: [
          { name: 'data-test', value: 'data-test-value' },
          { name: 'data-test-1', value: 'data-test-value-1' },
        ],
      });

      return expect(render(withExtraAttributes)).resolves.toMatchSnapshot();
    });
  });

  describe('Without translation', () => {
    test('renders correctly', () => {
      expect.assertions(1);

      return expect(render(dataWithoutTranslation)).resolves.toMatchSnapshot();
    });

    test('renders correctly with extra class names', () => {
      expect.assertions(1);

      const withExtraClasses = merge(dataWithoutTranslation, {
        extra_classes: 'custom-class custom-class--test',
      });

      return expect(render(withExtraClasses)).resolves.toMatchSnapshot();
    });

    test('renders correctly with extra attributes', () => {
      expect.assertions(1);

      const withExtraAttributes = merge(dataWithoutTranslation, {
        extra_attributes: [
          { name: 'data-test', value: 'data-test-value' },
          { name: 'data-test-1', value: 'data-test-value-1' },
        ],
      });

      return expect(render(withExtraAttributes)).resolves.toMatchSnapshot();
    });
  });
});
