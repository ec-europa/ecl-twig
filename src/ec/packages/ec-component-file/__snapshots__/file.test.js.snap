// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EC - File With translation renders correctly 1`] = `
<div
  class="ecl-file"
  data-ecl-file=""
>
  <div
    class="ecl-file__container"
  >
    <svg
      aria-hidden="true"
      class="ecl-icon ecl-icon--2xl ecl-file__icon"
      focusable="false"
    >
      <use
        xlink:href="/static/icons.svg#general--copy"
      />
    </svg>
    <div
      class="ecl-file__info"
    >
      <div
        class="ecl-file__title"
      >
        State of the Union 2018 brochure
      </div>
      <div
        class="ecl-file__language"
      >
        English
      </div>
      <div
        class="ecl-file__meta"
      >
        (16.2 MB - PDF)
      </div>
    </div>
    <a
      class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-after ecl-file__download ecl-button ecl-button--secondary"
      download=""
      href="/example"
    >
      <span
        class="ecl-link__label"
      >
        Download
      </span>
       
      <svg
        aria-hidden="true"
        class="ecl-icon ecl-icon--fluid ecl-link__icon"
        focusable="false"
      >
        <use
          xlink:href="/static/icons.svg#ui--download"
        />
      </svg>
    </a>
  </div>
  <div
    class="ecl-file__translation-container"
    data-ecl-file-translation-container=""
  >
    <button
      class="ecl-button ecl-button--ghost ecl-file__translation-toggle"
      data-ecl-file-translation-toggle=""
      type="button"
    >
      <span
        class="ecl-button__container"
      >
        <span
          class="ecl-button__label"
          data-ecl-label=""
        >
          Other languages (3)
        </span>
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--fluid ecl-icon--rotate-180 ecl-button__icon ecl-button__icon--after"
          focusable="false"
        >
          <use
            xlink:href="/static/icons.svg#ui--corner-arrow"
          />
        </svg>
      </span>
    </button>
    <ul
      class="ecl-file__translation-list"
    >
      <li
        class="ecl-file__translation-item"
      >
        <div
          class="ecl-file__translation-info"
        >
          <div
            class="ecl-file__translation-title"
            lang="bg"
          >
            български
          </div>
          <div
            class="ecl-file__translation-meta"
          >
            (15.7 MB - PDF)
          </div>
        </div>
        <a
          class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-after ecl-file__translation-download"
          download=""
          href="/example"
          hreflang="bg"
        >
          <span
            class="ecl-link__label"
          >
            Download
          </span>
           
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--fluid ecl-link__icon"
            focusable="false"
          >
            <use
              xlink:href="/static/icons.svg#ui--download"
            />
          </svg>
        </a>
      </li>
      <li
        class="ecl-file__translation-item"
      >
        <div
          class="ecl-file__translation-info"
        >
          <div
            class="ecl-file__translation-title"
            lang="es"
          >
            español
          </div>
          <div
            class="ecl-file__translation-meta"
          >
            (15.8 MB - PDF)
          </div>
        </div>
        <a
          class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-after ecl-file__translation-download"
          download=""
          href="/example"
          hreflang="es"
        >
          <span
            class="ecl-link__label"
          >
            Download
          </span>
           
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--fluid ecl-link__icon"
            focusable="false"
          >
            <use
              xlink:href="/static/icons.svg#ui--download"
            />
          </svg>
        </a>
      </li>
      <li
        class="ecl-file__translation-item"
      >
        <div
          class="ecl-file__translation-info"
        >
          <div
            class="ecl-file__translation-title"
            lang="fr"
          >
            français
          </div>
          <div
            class="ecl-file__translation-meta"
          >
            (15.4 MB - PDF)
          </div>
        </div>
        <a
          class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-after ecl-file__translation-download"
          download=""
          href="/example"
          hreflang="fr"
        >
          <span
            class="ecl-link__label"
          >
            Download
          </span>
           
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--fluid ecl-link__icon"
            focusable="false"
          >
            <use
              xlink:href="/static/icons.svg#ui--download"
            />
          </svg>
        </a>
      </li>
      <li
        class="ecl-file__translation-item ecl-file__translation-description"
      >
        Looking for another language which is not on the list? Find out why.
      </li>
    </ul>
  </div>
</div>
`;

exports[`EC - File With translation renders correctly with extra attributes 1`] = `
<div
  class="ecl-file"
  data-ecl-file=""
  data-test="data-test-value"
  data-test-1="data-test-value-1"
>
  <div
    class="ecl-file__container"
  >
    <svg
      aria-hidden="true"
      class="ecl-icon ecl-icon--2xl ecl-file__icon"
      focusable="false"
    >
      <use
        xlink:href="/static/icons.svg#general--copy"
      />
    </svg>
    <div
      class="ecl-file__info"
    >
      <div
        class="ecl-file__title"
      >
        State of the Union 2018 brochure
      </div>
      <div
        class="ecl-file__language"
      >
        English
      </div>
      <div
        class="ecl-file__meta"
      >
        (16.2 MB - PDF)
      </div>
    </div>
    <a
      class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-after ecl-file__download ecl-button ecl-button--secondary"
      download=""
      href="/example"
    >
      <span
        class="ecl-link__label"
      >
        Download
      </span>
       
      <svg
        aria-hidden="true"
        class="ecl-icon ecl-icon--fluid ecl-link__icon"
        focusable="false"
      >
        <use
          xlink:href="/static/icons.svg#ui--download"
        />
      </svg>
    </a>
  </div>
  <div
    class="ecl-file__translation-container"
    data-ecl-file-translation-container=""
  >
    <button
      class="ecl-button ecl-button--ghost ecl-file__translation-toggle"
      data-ecl-file-translation-toggle=""
      type="button"
    >
      <span
        class="ecl-button__container"
      >
        <span
          class="ecl-button__label"
          data-ecl-label=""
        >
          Other languages (3)
        </span>
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--fluid ecl-icon--rotate-180 ecl-button__icon ecl-button__icon--after"
          focusable="false"
        >
          <use
            xlink:href="/static/icons.svg#ui--corner-arrow"
          />
        </svg>
      </span>
    </button>
    <ul
      class="ecl-file__translation-list"
    >
      <li
        class="ecl-file__translation-item"
      >
        <div
          class="ecl-file__translation-info"
        >
          <div
            class="ecl-file__translation-title"
            lang="bg"
          >
            български
          </div>
          <div
            class="ecl-file__translation-meta"
          >
            (15.7 MB - PDF)
          </div>
        </div>
        <a
          class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-after ecl-file__translation-download"
          download=""
          href="/example"
          hreflang="bg"
        >
          <span
            class="ecl-link__label"
          >
            Download
          </span>
           
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--fluid ecl-link__icon"
            focusable="false"
          >
            <use
              xlink:href="/static/icons.svg#ui--download"
            />
          </svg>
        </a>
      </li>
      <li
        class="ecl-file__translation-item"
      >
        <div
          class="ecl-file__translation-info"
        >
          <div
            class="ecl-file__translation-title"
            lang="es"
          >
            español
          </div>
          <div
            class="ecl-file__translation-meta"
          >
            (15.8 MB - PDF)
          </div>
        </div>
        <a
          class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-after ecl-file__translation-download"
          download=""
          href="/example"
          hreflang="es"
        >
          <span
            class="ecl-link__label"
          >
            Download
          </span>
           
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--fluid ecl-link__icon"
            focusable="false"
          >
            <use
              xlink:href="/static/icons.svg#ui--download"
            />
          </svg>
        </a>
      </li>
      <li
        class="ecl-file__translation-item"
      >
        <div
          class="ecl-file__translation-info"
        >
          <div
            class="ecl-file__translation-title"
            lang="fr"
          >
            français
          </div>
          <div
            class="ecl-file__translation-meta"
          >
            (15.4 MB - PDF)
          </div>
        </div>
        <a
          class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-after ecl-file__translation-download"
          download=""
          href="/example"
          hreflang="fr"
        >
          <span
            class="ecl-link__label"
          >
            Download
          </span>
           
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--fluid ecl-link__icon"
            focusable="false"
          >
            <use
              xlink:href="/static/icons.svg#ui--download"
            />
          </svg>
        </a>
      </li>
      <li
        class="ecl-file__translation-item ecl-file__translation-description"
      >
        Looking for another language which is not on the list? Find out why.
      </li>
    </ul>
  </div>
</div>
`;

exports[`EC - File With translation renders correctly with extra class names 1`] = `
<div
  class="ecl-file custom-class custom-class--test"
  data-ecl-file=""
>
  <div
    class="ecl-file__container"
  >
    <svg
      aria-hidden="true"
      class="ecl-icon ecl-icon--2xl ecl-file__icon"
      focusable="false"
    >
      <use
        xlink:href="/static/icons.svg#general--copy"
      />
    </svg>
    <div
      class="ecl-file__info"
    >
      <div
        class="ecl-file__title"
      >
        State of the Union 2018 brochure
      </div>
      <div
        class="ecl-file__language"
      >
        English
      </div>
      <div
        class="ecl-file__meta"
      >
        (16.2 MB - PDF)
      </div>
    </div>
    <a
      class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-after ecl-file__download ecl-button ecl-button--secondary"
      download=""
      href="/example"
    >
      <span
        class="ecl-link__label"
      >
        Download
      </span>
       
      <svg
        aria-hidden="true"
        class="ecl-icon ecl-icon--fluid ecl-link__icon"
        focusable="false"
      >
        <use
          xlink:href="/static/icons.svg#ui--download"
        />
      </svg>
    </a>
  </div>
  <div
    class="ecl-file__translation-container"
    data-ecl-file-translation-container=""
  >
    <button
      class="ecl-button ecl-button--ghost ecl-file__translation-toggle"
      data-ecl-file-translation-toggle=""
      type="button"
    >
      <span
        class="ecl-button__container"
      >
        <span
          class="ecl-button__label"
          data-ecl-label=""
        >
          Other languages (3)
        </span>
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--fluid ecl-icon--rotate-180 ecl-button__icon ecl-button__icon--after"
          focusable="false"
        >
          <use
            xlink:href="/static/icons.svg#ui--corner-arrow"
          />
        </svg>
      </span>
    </button>
    <ul
      class="ecl-file__translation-list"
    >
      <li
        class="ecl-file__translation-item"
      >
        <div
          class="ecl-file__translation-info"
        >
          <div
            class="ecl-file__translation-title"
            lang="bg"
          >
            български
          </div>
          <div
            class="ecl-file__translation-meta"
          >
            (15.7 MB - PDF)
          </div>
        </div>
        <a
          class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-after ecl-file__translation-download"
          download=""
          href="/example"
          hreflang="bg"
        >
          <span
            class="ecl-link__label"
          >
            Download
          </span>
           
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--fluid ecl-link__icon"
            focusable="false"
          >
            <use
              xlink:href="/static/icons.svg#ui--download"
            />
          </svg>
        </a>
      </li>
      <li
        class="ecl-file__translation-item"
      >
        <div
          class="ecl-file__translation-info"
        >
          <div
            class="ecl-file__translation-title"
            lang="es"
          >
            español
          </div>
          <div
            class="ecl-file__translation-meta"
          >
            (15.8 MB - PDF)
          </div>
        </div>
        <a
          class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-after ecl-file__translation-download"
          download=""
          href="/example"
          hreflang="es"
        >
          <span
            class="ecl-link__label"
          >
            Download
          </span>
           
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--fluid ecl-link__icon"
            focusable="false"
          >
            <use
              xlink:href="/static/icons.svg#ui--download"
            />
          </svg>
        </a>
      </li>
      <li
        class="ecl-file__translation-item"
      >
        <div
          class="ecl-file__translation-info"
        >
          <div
            class="ecl-file__translation-title"
            lang="fr"
          >
            français
          </div>
          <div
            class="ecl-file__translation-meta"
          >
            (15.4 MB - PDF)
          </div>
        </div>
        <a
          class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-after ecl-file__translation-download"
          download=""
          href="/example"
          hreflang="fr"
        >
          <span
            class="ecl-link__label"
          >
            Download
          </span>
           
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--fluid ecl-link__icon"
            focusable="false"
          >
            <use
              xlink:href="/static/icons.svg#ui--download"
            />
          </svg>
        </a>
      </li>
      <li
        class="ecl-file__translation-item ecl-file__translation-description"
      >
        Looking for another language which is not on the list? Find out why.
      </li>
    </ul>
  </div>
</div>
`;

exports[`EC - File Without translation renders correctly 1`] = `
<div
  class="ecl-file"
  data-ecl-file=""
>
  <div
    class="ecl-file__container"
  >
    <svg
      aria-hidden="true"
      class="ecl-icon ecl-icon--2xl ecl-file__icon"
      focusable="false"
    >
      <use
        xlink:href="/static/icons.svg#general--copy"
      />
    </svg>
    <div
      class="ecl-file__info"
    >
      <div
        class="ecl-file__title"
      >
        State of the Union 2018 brochure
      </div>
      <div
        class="ecl-file__language"
      >
        English
      </div>
      <div
        class="ecl-file__meta"
      >
        (16.2 MB - PDF)
      </div>
    </div>
    <a
      class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-after ecl-file__download ecl-button ecl-button--secondary"
      download=""
      href="/example"
    >
      <span
        class="ecl-link__label"
      >
        Download
      </span>
       
      <svg
        aria-hidden="true"
        class="ecl-icon ecl-icon--fluid ecl-link__icon"
        focusable="false"
      >
        <use
          xlink:href="/static/icons.svg#ui--download"
        />
      </svg>
    </a>
  </div>
</div>
`;

exports[`EC - File Without translation renders correctly with extra attributes 1`] = `
<div
  class="ecl-file"
  data-ecl-file=""
  data-test="data-test-value"
  data-test-1="data-test-value-1"
>
  <div
    class="ecl-file__container"
  >
    <svg
      aria-hidden="true"
      class="ecl-icon ecl-icon--2xl ecl-file__icon"
      focusable="false"
    >
      <use
        xlink:href="/static/icons.svg#general--copy"
      />
    </svg>
    <div
      class="ecl-file__info"
    >
      <div
        class="ecl-file__title"
      >
        State of the Union 2018 brochure
      </div>
      <div
        class="ecl-file__language"
      >
        English
      </div>
      <div
        class="ecl-file__meta"
      >
        (16.2 MB - PDF)
      </div>
    </div>
    <a
      class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-after ecl-file__download ecl-button ecl-button--secondary"
      download=""
      href="/example"
    >
      <span
        class="ecl-link__label"
      >
        Download
      </span>
       
      <svg
        aria-hidden="true"
        class="ecl-icon ecl-icon--fluid ecl-link__icon"
        focusable="false"
      >
        <use
          xlink:href="/static/icons.svg#ui--download"
        />
      </svg>
    </a>
  </div>
</div>
`;

exports[`EC - File Without translation renders correctly with extra class names 1`] = `
<div
  class="ecl-file custom-class custom-class--test"
  data-ecl-file=""
>
  <div
    class="ecl-file__container"
  >
    <svg
      aria-hidden="true"
      class="ecl-icon ecl-icon--2xl ecl-file__icon"
      focusable="false"
    >
      <use
        xlink:href="/static/icons.svg#general--copy"
      />
    </svg>
    <div
      class="ecl-file__info"
    >
      <div
        class="ecl-file__title"
      >
        State of the Union 2018 brochure
      </div>
      <div
        class="ecl-file__language"
      >
        English
      </div>
      <div
        class="ecl-file__meta"
      >
        (16.2 MB - PDF)
      </div>
    </div>
    <a
      class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-after ecl-file__download ecl-button ecl-button--secondary"
      download=""
      href="/example"
    >
      <span
        class="ecl-link__label"
      >
        Download
      </span>
       
      <svg
        aria-hidden="true"
        class="ecl-icon ecl-icon--fluid ecl-link__icon"
        focusable="false"
      >
        <use
          xlink:href="/static/icons.svg#ui--download"
        />
      </svg>
    </a>
  </div>
</div>
`;
